### ISSUE TITLE
themes break global $PROMPT_COMMAND variables

### ISSUE THREAD

**Issue by BiRo70** [Bash-it/bash-it#731]
Hi, this is not a real issue, but I noticed that the history sharing between open terminals is broken by the theme's define "PROMPT_COMMAND=prompt_fuction" variable.

`$ echo $PROMPT_COMMAND`
`prompt`

as work around I moved to the end of .bashrc after loading bashit the line:
`export PROMPT_COMMAND="history -a; history -c; history -r; $PROMPT_COMMAND"`
Now it works at it should.

I'm just wondering if the themes should consider not overwriting such common variables like this one, but edit it instead.

Thanks

**Comment by nwinkler** [comment]
Good point - I'm not sure about the correct behavior. I for example quite like having a separate history per terminal. This allows me to have multiple projects/workspaces open and each one has their own command history.

Not sure whether we should change this. It might make sense to add a note in the documentation to show how to change this in case someone wants to use a common history for all terminal sessions.

**Comment by wyhfrank** [comment]
Actually fasd plugin wouldn't work neither, because of the same issue.

Then I found this in [fasd.plugin.bash](https://github.com/Bash-it/bash-it/blob/master/plugins/available/fasd.plugin.bash#L25):
```bash
    case $PROMPT_COMMAND in
      *_fasd_prompt_func*) ;;
      *) PROMPT_COMMAND="_fasd_prompt_func;$PROMPT_COMMAND";;
    esac
```
It might be good to do the same trick for the themes, too.

**Comment by sveneh** [comment]
Yes, stumbled upon this, too after enabling autojump, which also needs a modified `PROMPT_COMMAND` variable

The problem is the order how bash-it executes its customization. First it enables all plugins, then it sets the theme (in my case the _bobby_ theme) . And the theme overwrites `PROMT_COMMAND` without keeping any additions added by plugins.

In the autojump case it should be
`PROMPT_COMMAND='prompt_command ; autojump_add_to_database'`, but it's only
`PROMPT_COMMAND='prompt_command'

Two possible next steps

1. Setup themes before setting up plugins
2. If that's not possible, theme's cannot simply overwrite `PROMPT_COMMAND` but must keep additions that plugins add.

**Comment by nwinkler** [comment]
Ah, good catch! The order of loading things was recently changed (see #729 and #730) - it's a bit of a chicken/egg problem:

Themes might want to use functionality defined in the plugins, e.g. showing the battery status as part of the theme, which requires the `battery` plugin to be loaded first. Hence the plugins are now loaded first (see #730).

We didn't catch the `PROMPT_COMMAND` issue at that point, it looks like the themes need to be changed to not simply overwrite the `PROMPT_COMMAND` variable, but add to it instead.

I don't want to change the order back to themes first, since that change was done for a good reason. If someone is willing to provide a PR that updates the themes to not simply replace the `PROMPT_COMMAND` variable, I'll gladly merge the PR. Sounds like it would make sense to put a function for manipulating the `PROMPT_COMMAND` variable into the base theme. Similar to the `pathmunge` function, which is reused for adding to the path without duplicating values (https://github.com/Bash-it/bash-it/blob/13b643e93e28fcf96a58792e02393493e738d604/lib/helpers.bash#L411).

**Comment by nwinkler** [comment]
Fixed through #754 - please retry after updating to the latest version.

**Comment by nwinkler** [comment]
Thanks to @wyhfrank for the great job!
